### PR TITLE
ACMS-1490: PHPUnit testcases for API Dashboard - API Endpoints

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**
Fixes ACMS-1490

**Proposed changes**
PHPUnit testcases for API Dashboard - API Endpoints

### CODE REVIEW

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * Tests headless dashboard API Document.
+ *
+ * @group acquia_cms
+ * @group acquia_cms_headless
+ * @group medium_risk
+ * @group push
+ */
+class DashboardApiDocumentTest extends DashboardWebDriverTestBase {
+
+  use DashboardTableTrait;
```
Comment (vishalkhode1) [inline modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php:17]: @deepakmishra2 Please remove this trait if your test include testing with table. Same for below test as well

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    // Test API Documentation section exists, get API Documentation section.
+    $element = $assertSession->elementExists('css', $this->sectionSelector);
+
+    // Test OpenAPI Resources label exist.
+    $openApiResource = $element->find("css", '.headless-dashboard-openapi-resources > p > a');
+    $this->assertEquals('OpenAPI Resources', $openApiResource->getText());
+
+    // Test OpenAPI Resources link.
+    $buttonAction = $openApiResource->getAttribute('href');
```
Comment (vishalkhode1) [inline modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php:58]: @deepakmishra2 Also test if link has attribute target and value as _blank.
Screenshot 2023-01-24 at 10 32 48 PM

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php (hunk 1 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php:17]: +1

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php (hunk 2 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php:58]: +1, for testing OpenAPI Resource link, create separate test function if possible.

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiUrlTest.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * Tests headless dashboard API Url.
+ *
+ * @group acquia_cms
+ * @group acquia_cms_headless
+ * @group medium_risk
+ * @group push
+ */
+class DashboardApiUrlTest extends DashboardWebDriverTestBase {
+
+  use DashboardTableTrait;
```
Comment (rajeshreeputra) [inline modules/acquia_cms_headless/tests/src/Functional/DashboardApiUrlTest.php:17]: @deepakmishra2 Please remove this trait if your test include testing with table.

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiUrlTest.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+   * {@inheritdoc}
+   */
+  public function testButtons(): void {
+    $this->assertButton("Update Base API URL");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function testSection(): void {
```
Comment (rajeshreeputra) [inline modules/acquia_cms_headless/tests/src/Functional/DashboardApiUrlTest.php:46]: @deepakmishra2 instead testing the Base API Url in section can test as separate function if possible.
```
/**
   * Tests the base jsonapi URL.
   */
  public function testBaseApiUrl(): void {
    $element = $this->getSection()->find("css", ".headless-dashboard-api-url");
    $this->assertSame($element->getText(), "Base API Url: ". $this->baseUrl . "/jsonapi");
  }
```
  in testSection function you can test image exist or not, any thoughts? 
  cc @vishalkhode1

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php (hunk 1 above)
Comment (deepakmishra2) [reply modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php]: Done

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php (hunk 2 above)
Comment (deepakmishra2) [reply modules/acquia_cms_headless/tests/src/Functional/DashboardApiDocumentTest.php]: done

**Review by rajeshreeputra — APPROVED** [review]
Overall changes looks good.

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**